fem/fespace.hpp

```diff
@@ -473,28 +586,37 @@ class FiniteElementSpace
 
    int GetBdrAttribute(int i) const { return mesh->GetBdrAttribute(i); }
 
-   /// Returns indexes of degrees of freedom in array dofs for i'th element.
-   virtual void GetElementDofs(int i, Array<int> &dofs) const;
-
-   /// Returns indexes of degrees of freedom for i'th boundary element.
-   virtual void GetBdrElementDofs(int i, Array<int> &dofs) const;
-
-   /** @brief Returns the indexes of the degrees of freedom for i'th face
-       including the dofs for the edges and the vertices of the face. */
-   virtual void GetFaceDofs(int i, Array<int> &dofs) const;
-
-   /** @brief Returns the indexes of the degrees of freedom for i'th edge
-       including the dofs for the vertices of the edge. */
-   void GetEdgeDofs(int i, Array<int> &dofs) const;
+   /// Returns indices of degrees of freedom of element 'elem'.
+   virtual void GetElementDofs(int elem, Array<int> &dofs) const;
+
+   /// Returns indices of degrees of freedom for boundary element 'bel'.
+   virtual void GetBdrElementDofs(int bel, Array<int> &dofs) const;
+
+   /** @brief Returns the indices of the degrees of freedom for the specified
+       face, including the DOFs for the edges and the vertices of the face. */
+   /** In variable order spaces, multiple variants of DOFs can be returned.
+       See @a GetEdgeDofs for more details.
+       @return Order of the selected variant, or -1 if there are no more
+       variants.*/
+   virtual int GetFaceDofs(int face, Array<int> &dofs, int variant = 0) const;
+
+   /** @brief Returns the indices of the degrees of freedom for the specified
+       edge, including the DOFs for the vertices of the edge. */
+   /** In variable order spaces, multiple sets of DOFs may exist on an edge,
+       corresponding to the different polynomial orders of incident elements.
+       The 'variant' parameter is the zero-based index of the desired DOF set.
+       The variants are ordered from lowest polynomial degree to the highest.
+       @return Order of the selected variant, or -1 if there are no more
+       variants. */
```

Comment:
I don't think this is the right place to define the concept of `variant` which is central in all this new code, moreover after reading the documentation I still don't understand fully what it is.

Reply:
OK, I'm thinking of adding more documentation to the tables `var_edge_dofs` and `var_face_dofs`, where the "variant" concept could be better defined, maybe with an example too.

Reply:
Done, there is now a Doxygen comment describing the content of `var_edge_dofs` and `var_face_dofs`.